### DISCUSSION TITLE
Great project!

### DISCUSSION THREAD

**Discussion by pawamoy** [cduck/drawSvg#97] [General]
Hey @cduck, I've just started sponsoring you because your project is great :slightly_smiling_face: 
To be honest I did not use it yet, but am planning to, and upon looking at the issues/discussions/examples, it seems to me that the code was very well designed, and your interactions with the community are very enjoyable.

About my use of drawsvg: I'm trying to document one of my project by including diagrams, to help users understand what's going on. I mainly need to show graph-like diagrams, probably with animations showing the different steps taken. I feel like I searched the entire web for a library that would match my requirements, and yours is the **only one** that gets very close:

- [ ] high-level: it's too low-level for my use-case, but that's fine, I should be able to build upon it :slightly_smiling_face: 
- [x] pure python (no DSL, binary executable or system libs needed)
- [x] no dependencies :heart: :rocket: (too many of the alternatives have a load of dependencies)
- [x] renders SVG directly
- [x] flexible: allows placements of objects. I could have stayed on Mermaid if it wasn't so bad at placing nodes :sweat_smile: 
- [x] library-first (not forcing me to patch things like other projects)

The closest alternative was [Chalk](https://github.com/chalk-diagrams/chalk). It's a bit higher level but doesn't check many of the other boxes.

I'm not sure when I'll have time to play with drawsvg, but I'm sure it will be fun! Thanks for the amazing project!

**Comment by pawamoy** [comment]
By the way, nice logo, I thought I was going crazy :rofl:

**Comment by cduck** [comment]
🎉

Thanks for the sponsorship!

> high-level: it's too low-level for my use-case, but that's fine, I should be able to build upon it 🙂

When you do try out drawsvg, please share your thoughts!  I'd welcome any ideas for higher level constructs that integrate well with the rest of the library or even for a new library.  Hyperbolic is one example of a high-level (but domain-specific) drawing library built on drawsvg.  https://github.com/cduck/hyperbolic

I've also seen this cool project but I haven't tried it out yet: https://github.com/joelle-o-world/pattern-cutting

**Reply by pawamoy** [reply]
Of course, I'll share my experience with drawsvg with you :slightly_smiling_face: 
Thanks, I added pattern-cutting to the list of things to add to my Markdown Exec gallery :smile: https://github.com/pawamoy/markdown-exec/issues/9
